src/esp/assets/Attributes.h

```diff
@@ -273,12 +296,8 @@ class AbstractPrimitiveAttributes : public esp::core::Configuration {
   // virtual ~AbstractPrimitiveAttributes() = 0;
 
   // originHandle is set internally based on attributes configuration
-  std::string getOriginHandle() const { return getString("originHandle"); }
-
-  void setAssetTemplateID(int assetTemplateID) {
-    setInt("assetTemplateID", assetTemplateID);
-  }
-  int getAssetTemplateID() const { return getInt("assetTemplateID"); }
+  // setting externally is prohibited
+  void setOriginHandle(const std::string&) {}
```

Comment:
It is empty, and it is not a virtual function. 
If user cannot set it externally, can we remove this function?

Reply:
This is correct.  It will become virtual once the appropriate pybinds are developed.  For pybind to support virtual, seems you have to get "trampoline" classes implemented in c++ and that was beyond the scope of this PR.

Reply:
To elaborate more on this and other similar issues in the Attributes : 

All of the various attributes classes have the primary purpose of providing the user with easily accessible editing capabilities for the objects they describe, and of course the key mechanism for this access is pybind.  The integration of pybind requires, in some cases, to build c++ "trampoline" classes that act as adapters between the underlying c++ code and pybind's python interface/code gen.  These classes are required by pybind, for example, to implement proper support of virtual functions.  

The inheritance hierarchy of the attributes is intended to minimize code duplication and provide code expectations (much like an interface) to future attributes class development.  As an aside, ideally, all the attributes classes that are named "AbstractXXX" would actually -be- abstract, but since they inherit from a concrete class (configuration), it is not immediately obvious how to properly accomplish this in pybind (I am still researching how to do it, but I have not yet figured out the mechanism).  I understand that this kind of inheritance hierarchy is uncommon, but since configuration is in an external library/used many places elsewhere, I am stuck with using it as an instantiable class.  As far as the entire "Attributes" subtree of functionality, however, the current hierarchy satisfies Liskov substitution. 

However, pybind's seemingly arcane mechanisms to accomplish this inheritance hierarchy has put me in the undesirable position of having to slightly modify the c++ in suboptimal ways to accommodate pybind itself, at least temporarily.  Thankfully, these modifications have been slight, but the code above is an example of one of them.

The next PR will introduce the necessary trampoline classes required to support the virtual functions in the various attributes classes.  The purpose of this PR was primarily to introduce the concept of the AttributesManagers, and as you said, it is a pretty large PR as it is, and so I felt that making suboptimal (but temporary) coding decisions as above was preferable to substantially increasing the size of the PR itself with the trampolines :).

Reply:
I'll check that out, thanks.